File: java/src/Ui/UserInput.java
package Ui;

class UserInput {

	char command;
	String[] args;

	UserInput(String line) {
		command = line.charAt(0);
		args = line.split("\\s+");
	}

	public String args(int i) {
		if (i < 0)
			throw new IllegalArgumentException();
		return (i < args.length) ? args[i] : null;
	}

}

File: dotnet/Ui/UserInput.cs
namespace Ui 
{
	using System;

	public class UserInput {

		internal char command;
		public string[] args;

		internal UserInput(string line) {
			command = line[0];
			args = line.Split(null);
		}

		public string Args(int i) {
			if (i < 0)
				throw new IndexOutOfRangeException();
			return (i < args.Length) ? args[i] : null;
		}

	}
}